connectors/src/connectors/zendesk/index.ts

```diff
@@ -123,8 +144,145 @@ export class ZendeskConnectorManager extends BaseConnectorManager<null> {
   }: {
     permissions: Record<string, ConnectorPermission>;
   }): Promise<Result<void, Error>> {
-    logger.info({ permissions }, "Setting permissions");
-    throw new Error("Method not implemented.");
+    const connector = await ConnectorResource.fetchById(this.connectorId);
+    if (!connector) {
+      logger.error(
+        { connectorId: this.connectorId },
+        "[Zendesk] Connector not found."
+      );
+      return new Err(new Error("Connector not found"));
+    }
+
+    const connectionId = connector.connectionId;
+    const zendeskConfiguration = await ZendeskConfigurationResource.fetchById(
+      this.connectorId
+    );
+    if (!zendeskConfiguration) {
+      logger.error(
+        { connectorId: this.connectorId },
+        "[Zendesk] ZendeskConfiguration not found. Cannot set permissions."
+      );
+      return new Err(new Error("ZendeskConfiguration not found"));
+    }
+    const subdomain = zendeskConfiguration.subdomain;
+
+    const toBeSignaledBrandIds = new Set<number>();
+    const toBeSignaledTicketsIds = new Set<number>();
+    const toBeSignaledHelpCenterIds = new Set<number>();
+    const toBeSignaledCategoryIds = new Set<number>();
+    try {
```

Comment:
IMHO we shouldn't have this "catch all" try/catch. If the error is an unexpected exception, you should let it throw. The endpoint will 500 which is what we want.
We return `Err` when the error is "expected".